go.mod

```diff
@@ -14,7 +14,7 @@ require (
 	github.com/hashicorp/logutils v1.0.0
 	github.com/hashicorp/terraform-exec v0.20.0
 	github.com/hashicorp/terraform-json v0.21.0
-	github.com/hashicorp/terraform-plugin-go v0.21.0
+	github.com/hashicorp/terraform-plugin-go v0.22.0
```

Comment:
Running `go get github.com/hashicorp/terraform-plugin-go` results in the addition of the following entry to `go.mod`:

```
toolchain go1.21.6
```

Which would result in an analogous [CI failure](https://github.com/hashicorp/terraform-plugin-mux/actions/runs/8017456549/job/21901361119#step:4:7) to that seen on _terraform-plugin-mux_

Given that our CI runs both Go `1.20` and `1.21`, I removed this directive, but perhaps this should be handled differently?